tasks/src/main/scala/dagr/tasks/misc/BwaMemStreamed.scala

```diff
@@ -53,48 +54,25 @@ class BwaMemStreamed(unmappedBam: PathToBam,
                      mergeBamAlignmentMem: String = "2G",
                      fifoBufferMem: String = "512M",
                      processAltMappings : Boolean = true
-                    ) extends PipedTask with VariableResources {
-  this.name = "BwaMemStreamed"
-  override protected val tasks = ListBuffer[ProcessTask]()
+                    ) extends Pipeline {
```

Comment:
I think this is a little complicated.  When we want to encapsulate a set of streamed tasks I think there are three options:
1. extend pipeline (as here).  You could, in theory, extend `Piping` and implement those methods to vector to the internally held piping object, but I think that violates some of our notions about depending across pipeline boundaries.
2. extend Task directly, extend 'Piping' and implement vectoring methods.  Basically the same as 1, but without violating `Pipeilne`'s contract
3. Don't have a class at all, but just an object with an apply method that returns the pipe chain.

The third option seems best to me now that I think about it.